scim-spec/scim-spec-schema/src/main/java/org/apache/directory/scim/spec/filter/InMemoryScimFilterMatcher.java

```diff
@@ -158,8 +159,11 @@ public boolean test(R actual) {
 
           // now walk the attribute path again to get the accessor and value
           Schema.Attribute schemaAttribute = attributeContainer.getAttribute(attributeReference.getAttributeName());
-          actual = schemaAttribute.getAccessor().get(actual);
-
+          if (actual instanceof ScimResource) {
+            // actual is the top level scim resource - need to extract the top attribute.
+            // otherwise we can move on directly to the sub-attribute
+            actual = schemaAttribute.getAccessor().get(actual);
+          }
```

Comment:
@erant10 
🤔 This bit has me confused.
Locally, I reverted this back the previous try/catch/todo, and all the tests pass.

But I haven't been able to figure out what condition causes this error. Any ideas?